src/codegen/generators/expression_generator.rs

```diff
@@ -215,7 +215,15 @@ impl<'a, 'b> ExpressionCodeGenerator<'a, 'b> {
 
                 let index_entry = self.index.get_type(call_name.unwrap())?;
                 Ok((callable_reference, index_entry))
-            }
+            },
+            Statement::QualifiedReference { .. } => {
+                let loaded_value = self.generate_load_for(operator);
```

Comment:
don't you hide the original error here?
can you just error-propagate the load-call and see what the original problem is?

```suggestion
                let type_and_ptr = self.generate_load_for(operator)?;
                (type_and_ptr.ptr_value, type_and_ptr.type_entry)
```